src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -94,6 +112,122 @@ void RuntimeConfigurator::init_data_info(const std::shared_ptr<lowered::LinearIR
     }
 }
 
+void RuntimeConfigurator::init_buffer_info(const std::shared_ptr<lowered::LinearIR>& linear_ir) {
```

Comment:
Ideally all this work should be performed in the constructor of a dedicated transformation, and the configurator should simply instantiate all these transformation on inialization stage.
And on update stage, these transformations should simply be executed on lir.
This is a more scalable and extendable approach compared to the preset one.

We shouldn't do it now, but it seems like the right direction to move to. What do you think?